js/viz/chart.js

```diff
@@ -371,31 +371,38 @@ function collectMarkersInfoBySeries(allSeries, filteredSeries, argAxis) {
     return { series, overloadedSeries };
 }
 
-function applyAutoHidePointMarkers(allSeries, filteredSeries, overloadedSeries, argAxis) {
-    const argAxisType = argAxis.getOptions().type;
-    filteredSeries.forEach(s => {
-        const seriesIndex = allSeries.indexOf(s);
-        s.autoHidePointMarkers = false;
-        const tickCount = argAxis.getTicksValues().majorTicksValues.length;
-        if(s.autoHidePointMarkersEnabled() && (argAxisType === DISCRETE || overloadedSeries[seriesIndex].pointsCount > tickCount)) {
-            for(const index in overloadedSeries[seriesIndex]) {
-                const i = parseInt(index);
-                if(isNaN(i) || overloadedSeries[seriesIndex].total / overloadedSeries[seriesIndex].continuousSeries < 3) {
-                    continue;
-                }
-                if(i === seriesIndex) {
-                    if(overloadedSeries[i][i] * 2 >= overloadedSeries[i].pointsCount) {
-                        s.autoHidePointMarkers = true;
-                        break;
-                    }
-                } else if(overloadedSeries[seriesIndex].total >= overloadedSeries[seriesIndex].pointsCount) {
-                    s.autoHidePointMarkers = true;
-                    break;
-                }
+const checkOverlay = (overlayPoint, currentPoint, radiusPoint) => {
+    const leftBoundaryOverlay = overlayPoint.x - radiusPoint <= currentPoint.x;
+    const rightBoundaryOverlay = overlayPoint.x + radiusPoint >= currentPoint.x;
+    const topBoundaryOverlay = overlayPoint.y - radiusPoint <= currentPoint.y;
+    const bottomBoundaryOverlay = overlayPoint.y + radiusPoint >= currentPoint.y;
+
+    const isOverlayX = leftBoundaryOverlay && rightBoundaryOverlay;
```

Comment:
What does it mean `isOverlayX`? How point can be an `overlay`? Can we make this variables' name more specific?

Reply:
When two points intersect visually, one point overlaps with the second